### PR TITLE
{devel} [all of them, yes, ALL of them] enable ncursesw in all ncurses easyconfigs (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
Started this for #1327

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, AMD Opteron(tm) Processor 6136, Python 2.6.6
See https://gist.github.com/7f9e6619a34ab279d7bd for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/837644fdb72bb755e872 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, AMD Opteron(tm) Processor 6136, Python 2.6.6
See https://gist.github.com/b068531dc9ed4eece54e for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/0caf5eea7ec28f0c5d4d for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/f37a6903adcbe60c7702 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/e50ee87ed35d46d43277 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/26941e138601d2289d7a for a full test report.

**Comment by wpoely86** [comment]
@boegel I think this is ready, you can give it a spin. Maybe @fgeorgatos too?

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9_configure_darwin.patch [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# in relation to -no-cpp-precomp bug: https://lists.gnu.org/archive/html/bug-ncurses/2012-12/msg00001.html
+# by Fotis
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9_configure_darwin.patch]: tried/proposed by Fotis, meant for MacOSX compatible nurses builds...

overall comment for this PR: we sorely need next-generation easyconfigs to consolidate the information!

**Comment by wpoely86** [comment]
@boegel should be good to go

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c3bc40f7f15d16c30f83 for a full test report.

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-GCC-4.7.2.eb [hunk 2]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-GCC-4.7.2.eb]: patch doesn't work on this version...

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmpolf-1.1.6.eb [hunk 3]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmpolf-1.1.6.eb]: this patch doesn't work on this version

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmvolf-1.1.12rc1.eb [hunk 4]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmvolf-1.1.12rc1.eb]: this patch doesn't work on this version

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmvolf-1.2.7.eb [hunk 5]
```diff
@@ -9,12 +11,31 @@ function-key mapping, and has all the other SYSV-curses enhancements over BSD Cu
 toolchain = {'name': 'cgmvolf', 'version': '1.2.7'}
 toolchainopts = {'optarch': True, 'pic': True}
 
-sources = [SOURCE_TGZ]
 # In future, this URL will not be sufficient to obtain this tarball, because
 # the directory contains only a couple of older ncurses 5.9 snapshots.
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
+sources = [SOURCE_TGZ]
+
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgmvolf-1.2.7.eb]: this patch doesn't work on this version

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgoolf-1.1.7.eb [hunk 6]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-cgoolf-1.1.7.eb]: this patch doesn't work on this version

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-gmvolf-1.7.12.eb [hunk 7]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-gmvolf-1.7.12.eb]: this patch doesn't work on this version

File: easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-gmvolf-1.7.12rc1.eb [hunk 8]
```diff
@@ -14,7 +16,26 @@ toolchainopts = {'optarch': True, 'pic': True}
 source_urls = ['ftp://invisible-island.net/ncurses/current/']
 sources = [SOURCE_TGZ]
 
+patches = ['ncurses-5.9_configure_darwin.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-5.9-20130406-gmvolf-1.7.12rc1.eb]: this patch doesn't work on this version

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9b8d666fb46bed86fcda for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/00123e532d4b3a0500e5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/c9055c995234b809fdaa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 30 out of 54 (44 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e0e8f53add86a8ff7f92 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 30 out of 44 (44 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/096b3ea5237f1be890a2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 61 out of 61 (44 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e23e88f537135387efd0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/6e95f598b321fa730152 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/a28ffd18c7beb24ecadc for a full test report.

**Comment by boegel** [comment]
retested on top of https://github.com/hpcugent/easybuild-framework/pull/1294 to fix the issues with having various toolchains in play in a single session, all good now!

Thanks @wpoely86 for diving in!

**Pull request merged**